src/tools/schema2kotlin.ts

```diff
@@ -144,11 +151,13 @@ class KotlinGenerator implements ClassGenerator {
   fieldSerializes: string[] = [];
   fieldDeserializes: string[] = [];
   fieldsForToString: string[] = [];
+  singletonSchemaFields: string[] = [];
+  collectionSchemaFields: string[] = [];
 
   constructor(readonly node: SchemaNode, private readonly opts: minimist.ParsedArgs) {}
 
   // TODO: allow optional fields in kotlin
-  addField(field: string, typeChar: string, isOptional: boolean, refClassName: string|null) {
+  addField(field: string, typeChar: string, isOptional: boolean, refClassName: string|null, isCollection: boolean = false) {
```

Comment:
This is growing a lot of params (same for other files with this method above), I've seen code using interfaces to define required/optional, could consider now or for later, here's an example: https://github.com/PolymerLabs/arcs/blob/master/src/runtime/arc.ts#L64

Reply:
I like this suggestion. Implemented.